### ISSUE TITLE
Unable to initialize shared memory key

### ISSUE THREAD

**Issue by david-waterworth** [triton-inference-server/server#6195]
**Description**
I have two models, one using the `python` runtime and the other `pytorch` (`onnxruntime`)

My machine has 2 3090RTX GPUs.

When I try and start the triton docker container and run 

```
# tritonserver --model-repository=/mnt/model-repository
```
It fails `UNAVAILABLE: Internal: Unable to initialize shared memory key 'triton_python_backend_shm_region_3' to requested size (67108864 bytes). If you are running Triton inside docker, use '--shm-size' flag to control the shared memory region size. Each Python backend model instance requires at least 64MBs of shared memory. Error: No space left on device `

In the log I can also see

| cuda_memory_pool_byte_size{0}    | 67108864                                                                                                                                                                                                       
| cuda_memory_pool_byte_size{1}    | 67108864  

67108864 == 64MB

I fixed it using '--shm-size=128m` but I'm wondering if this is expected - i.e. the error seems to indicate that 64MB should be enough and presumably is shared between the 2 GPUs and the CPU.

`--shm-size=65m` and  '--shm-size=127m` failed - so I'm guessing I need 64m per GPU but that's not how I read the error message?

I don't have nvlink enabled, for some reason I've never managed to get it to work on my system.

**Triton Information**
2.36.0

**Comment by rmccorm4** [comment]
Hi @david-waterworth,

What is your `instance_group` settings in your python model config?

The 64MB requirement is per model **instance**, so if you have a python model with 2 instances, you'd need `--shm-size` or at least `128MB`. The `cuda_memory_pool_byte_size` is unrelated to the python backend shared memory size.

**Comment by david-waterworth** [comment]
The  `python` model config has:

```
instance_group [
    {
      count: 1
      kind: KIND_GPU
    }
]
```

The `pytorch` model config doesn't have an `instance_group` set - is KIND_GPU the default?

I forgot to mention there's also an ensemble model that orchestrates both - that doesn't set `instance_group` either.

(I'm using this example - https://blog.ml6.eu/triton-ensemble-model-for-deploying-transformers-into-production-c0f727c012e3 - but I appear to have dropped  `instance_group`  from the pytorch model config).

Note though, I only have the issue on my home 2xGPU machine, my work machine has a single 2090 GPU and works fine with 64MB of shared memory. I'm at work, I'll test when I get home if adding instance_group to the pytorch models helps (also I did get nvlink working - somethings changed since last time I tried, probably a newer driver)
